File: dotnet/UrlShortener/Controllers/EncodeController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using UrlShortener.Dtos;

namespace UrlShortener.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class EncodeController : ControllerBase
    {
        private readonly ILogger<EncodeController> _logger;

        public EncodeController(ILogger<EncodeController> logger)
        {
            _logger = logger;
        }

        [HttpPost]
        public ShortenedUrlDto Encode(UrlDto request) => throw new System.NotImplementedException();
    }
}

File: nestjs/src/controllers/encode.controller.ts
import {
  Body,
  Controller,
  NotImplementedException,
  Post,
} from '@nestjs/common';
import { ShortenedUrlDto } from '../dtos/shortenedUrl.dto';
import { UrlDto } from '../dtos/Url.dto';

@Controller()
export class EncodeController {
  @Post('/encode')
  encode(@Body() urlDto: UrlDto): ShortenedUrlDto {
    throw new NotImplementedException();
  }
}
